File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currSum = 0;
        int MaxSum = Integer.MIN_VALUE;

        for(int val : nums){
            currSum += val;
            MaxSum = Math.max(currSum , MaxSum);
        
        if(currSum < 0){
            currSum = 0;
        }
        }
        return MaxSum;
    }
}
    

File: Data Structures & Algorithms/non-cyclical-number/submission-0.cpp
class Solution {
public:
    bool isHappy(int n) {
       vector<int> nums(1001 , 0 ) ; 
        

        while( n != 1 && nums[n] != 1 ) { 
            nums[n] = 1  ; 
            int temp = n ; 
            n=0 ; 
            while( temp != 0 ) {
              int p = temp %10 ; 
              n+= (p*p) ; 
              temp /= 10  ; 

            }

            // cout<< n << " " ; 

            
             
        }
        return (n==1) ;  
    }
};
